### PR TITLE
ci: add helm install/upgrade regression test pipeline

### PR DESCRIPTION
**Pull request by yangchiu**
ci: add helm install/upgrade regression test pipeline

For https://github.com/longhorn/longhorn/issues/3987

Signed-off-by: Yang Chiu <yang.chiu@suse.com>

### CODE REVIEW

File: jenkins-jobs/longhorn-helm-chart-test.yml [hunk 1]
```diff
@@ -0,0 +1,137 @@
...
+              kubernetes version that will be deployed
+              for rke2: (default: v1.27.2+rke2r1)
+              for k3s: (default: v1.27.1+k3s1)
+      - string:
+          name: DISTRO
+          default: "sles"
+          description: "Linux distro used to install on created instances, supported values: [ubuntu, rhel, sles, centos, oracle, rockylinux, sle-micro]"
+      - string:
+          name: DISTRO_VERSION
+          default: "15-sp4"
```
Comment (innobead) [inline jenkins-jobs/longhorn-helm-chart-test.yml:107]: Can be 15-sp5 now, it seems it has been released.

**Review by innobead — CHANGES_REQUESTED** [review]
LGTM, just one distro version comment.

File: jenkins-jobs/longhorn-helm-chart-test.yml (hunk 1 above)
Comment (yangchiu) [reply jenkins-jobs/longhorn-helm-chart-test.yml:107]: This can be tested first, and then update this value in all pipelines together in another PR.

Comment (innobead) [reply jenkins-jobs/longhorn-helm-chart-test.yml:107]: Sounds good. 

Create a ticket to update this generally and we can also update the distro matrix for 1.5. (No need to do older versions even they are still supported like 1.4 or 1.3)

P.S. usually we only update the support matrix during a new feature release, but it seems sp5 just released after 1.5.0, so this is an exception.

cc @longhorn/qa

**Review by innobead — APPROVED** [review]

**Pull request merged**